targets/simple_switch/main.cpp

```diff
@@ -20,15 +20,81 @@
 
 /* Switch instance */
 
+#ifdef BM_HAVE_DLOPEN
```

Comment:
is there actually a need for 2 separate processor flags here, or based on the configure `BM_ENABLE_MODULES` => `BM_HAVE_DLOPEN` (otherwise configure would fail).

Reply:
This was in preparation in case another method other than dlopen would allow this same behaviour, e.g. Windows' LoadLibrary.

I can simplify if you prefer.

Reply:
no need, sounds good